staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -181,6 +186,11 @@ func (s *EtcdOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.EncryptionProviderConfigFilepath, "encryption-provider-config", s.EncryptionProviderConfigFilepath,
 		"The file containing configuration for encryption providers to be used for storing secrets in etcd")
 
+	fs.BoolVar(&s.EncryptionProviderConfigAutomaticReload, "encryption-provider-config-automatic-reload", s.EncryptionProviderConfigAutomaticReload,
+		"Determines if the file set by --encryption-provider-config should be automatically reloaded if the disk contents change. "+
+			"Setting this to true disables the ability to uniquely identify distinct KMS plugins via the API server healthz endpoints. "+
+			"If any KMS v2 plugins are used, this flag is ignored and the encryption config is always automatically reloaded, and the healthz endpoints are always merged.")
```

Comment:
making the reload behavior rely on content in the thing being reloaded is weird

I think this should limit itself to saying that enabling reloading collapses all kms providers to a single healthz check

separately (in the kmsv1 and kmsv2 config docs?) say that individual kms healthz checks are only provided when kmsv1 impls are in use and the encryption config is not auto-reloaded

Reply:
Opened #113631 to track updating docs around this.